### ISSUE TITLE
ascent grade vs route grade in stats / logbook

### ISSUE THREAD

**Issue by nicHoch** [theCrag/website#1894]
in the profile stats and fassette search the route grade dominates the ascent grade. 
This might lead to "false" results:
1. if the user logs only some pitches of a multipitch route (duplicate of #1698)
2. if the user logs in the ascent an different personal grade as the route in the index

**Comment by brendanheywood** [comment]
I think we need to have a broader discussion around ascent grades and resolve a few things that also overlap. 
* Like I would prefer in most cases to have my ascent grades change if the route grade changes, but this doesn't make sense in all cases.
* A similar thing with gym routes which don't yet have a grade.
* Log an ascent grade as a 'suggested grade' when you tick a route. How does this relate to a contributed grade for the route, these are really the same thing??

So would like to flesh out all the various use cases and then make some calls on what we really care about, and balance that with keeping the ascent form simple.

**Comment by scd** [comment]
I think we should adopt the same text based grade as per changes to
registered grade.  Maybe we could have a toggle which turned static text
improve an input field. If the static text is used then the ascent is kept
in sync with the route, otherwise if the input text box is used it is
managed independently of the route.

Should we resolve whole grades first? I think so.

**Comment by brendanheywood** [comment]
+1 to both ideas. The tricky bit is if the grade is overridden, how do we track whether it is a 'this isn't 23 it's a 22' vs a 'I went off route and what I did was 22'. I'd like to use the former to make real consensus grades or at least recommendations but the latter should be filtered.

Whole grade I would simple mean they get to choose which whole grade to use when they log the ascent. ie it could auto choose if they are french and their is both a french and au grade in thailand. Or if there is a free and aid whole grade, we could try and determine this based on the tick type.

**Comment by scd** [comment]
If somebody writes something different then we could simply add a checkbox and ask them if they want to record this grade as their opinion of the the routes grade. This would then update the route grade contributions explicitly, but not change the registered grade.

**Comment by brendanheywood** [comment]
Yes the challenge here is just making the UI succinct in an already cluttered form for a non-primary use case

**Comment by birgander2** [comment]
A nice feature some other sites are offering is a histogram of the ascent grades. Could be shown beside the "route quality" and "lead styles"!? Personally, I would not override the "topo" grade automatically.

**Comment by Drazhar** [comment]
I like the solution from 27crags for this topic. An example:
http://27crags.com/crags/magic-wood/routes/moderballett

So I like the idea, that the logged ascent grade is automatically used as a grad contribution and you get a graph if there are differing opinions. If there are a lot differing opinions, the route grade will be adjusted automatically, which is at some point ok.
The only problem I see with this is, that 80% of the people who are logging ascents don't really adjust the grade and because of that, this isn't that convincing.

**Comment by dnnr** [comment]
It seems like ever since #1698 has been solved, this issue has become more pressing.
Since being able to log multipitches properly, false entries in the logbook view have become a real annoyance. I've climbed in a few areas where a lot of routes not actual multipitches but split into a first and usually way harder second part, with an anchor inbetween.

So looking at my logbook now, I see things like this:
![te-7euttcqaf2ast](https://user-images.githubusercontent.com/2630359/38195559-4a50ad32-367e-11e8-9e20-2d37d94909f7.png)
But I haven't flashed a 6c+ on that day. I did a 5a, which is the grade of the first pitch. The details page confirms this:
![blwq1wpdaxypcyzi](https://user-images.githubusercontent.com/2630359/38195635-bacf5630-367e-11e8-822a-8855412d55e5.png)

So while I get that there are more complex aspects to this issue (such as how to deal with grades changing after ticking). But I'd argue that specifically this scenario is an obvious bug with an obvious fix, isn't it?

In my opinion, when listing ascents in the logbook, showing the ascent grade is invariably more correct than showing the route grade. After all, the items being shown are the *ascents* not *routes*.
